### PR TITLE
Feat/error views

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

The 404 page returned by default is really... minimalist
#1217 

## Proposal

- [x] Create custom views to handle errors (400, 403, 404, 500)

aperçu

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,8 +8,12 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unrealeased]
 
-### Changed
+### Add
```
Comment (jmaupetit) [inline CHANGELOG.md:11]:
```suggestion
### Added

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,8 +8,12 @@ Versioning](https://semver.org/spec/v2.0.0.html).
...
+- Use custom views to handle errors (400, 403, 404, 500)
 
+### Changed
```
Comment (jmaupetit) [inline CHANGELOG.md:14]:
```suggestion
### Changed

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,8 +8,12 @@ Versioning](https://semver.org/spec/v2.0.0.html).
...
+- Spread body-content block if document content height is smaller than
+  the window height.
```
Comment (jmaupetit) [inline CHANGELOG.md:16]:
```suggestion
  the window height
```

File: src/richie/apps/core/templates/richie/error.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+{% block head_title %}
+    {{status}} - {{ title }}
+{% endblock head_title %}
+
+{% block breadcrumbs %}{% endblock breadcrumbs %}
+
+
+{% block content %}
+<div class="error">
+  <svg class="error-icon"><use xlink:href="#icon-search-fail" /></svg>
```
Comment (jmaupetit) [inline src/richie/apps/core/templates/richie/error.html:13]: Didn't know about this `use` SVG tag. Have you seen this deprecation message? https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href

File: src/richie/apps/core/views/error.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    return render(
+        request,
+        template_name="richie/error.html",
+        status=status,
+        context={
+            "error": exception,
+            "status": status,
+            "title": CONTEXT_ERRORS[status]["title"],
+            "content": CONTEXT_ERRORS[status]["content"],
```
Comment (jmaupetit) [inline src/richie/apps/core/views/error.py:39]: What if you receive an unexpected status?

File: sandbox/urls.py [hunk 6]
```diff
@@ -62,3 +62,8 @@
         + staticfiles_urlpatterns()
         + urlpatterns
     )
+
+handler400 = "richie.apps.core.views.error.error_400_view_handler"
+handler403 = "richie.apps.core.views.error.error_403_view_handler"
+handler404 = "richie.apps.core.views.error.error_404_view_handler"
+handler500 = "richie.apps.core.views.error.error_500_view_handler"
```
Comment (jmaupetit) [inline sandbox/urls.py:69]: I think it's worth be documented somewhere? In an integration guide?

File: src/richie/apps/core/views/error.py (hunk 5 above)
Comment (jmaupetit) [reply src/richie/apps/core/views/error.py:39]: That said, it shouldn't happen...

File: src/richie/apps/core/templates/richie/error.html (hunk 4 above)
Comment (jbpenrath) [reply src/richie/apps/core/templates/richie/error.html:13]: I missed that. Good catch, we have to update all your templates.

File: sandbox/urls.py (hunk 6 above)
Comment (jbpenrath) [reply sandbox/urls.py:69]: Hmm this is a django feature.

File: src/richie/apps/core/views/error.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        "content": _("The requested resource does not exists."),
+    },
+    500: {
+        "title": _("Server bad request"),
+        "content": _("Something went wrong. Please retry later."),
+    },
+}
+
+
+def error_view_handler(request, exception, status):
```
Comment (sampaccoud) [inline src/richie/apps/core/views/error.py:27]: I think you should add a few tests on this view.

File: sandbox/urls.py (hunk 6 above)
Comment (jmaupetit) [reply sandbox/urls.py:69]: True. But I wish there was an integration guide for beginners...

File: CHANGELOG.md (hunk 1 above)
Comment (jmaupetit) [reply CHANGELOG.md:11]: Doesn't seem to be fixed :wink:

File: CHANGELOG.md (hunk 3 above)
Comment (jmaupetit) [reply CHANGELOG.md:16]: Ditto.

File: src/richie/apps/core/templates/richie/error.html (hunk 4 above)
Comment (jmaupetit) [reply src/richie/apps/core/templates/richie/error.html:13]: You'll handle this in another PR?

Comment (jbpenrath) [reply src/richie/apps/core/templates/richie/error.html:13]: Yes, it's clearer

Comment (jbpenrath) [reply src/richie/apps/core/templates/richie/error.html]: #1234

File: sandbox/urls.py (hunk 6 above)
Comment (jbpenrath) [reply sandbox/urls.py:69]: You are right. This is one of the next big steps, I noted to address this aspect.
#1235

File: src/richie/apps/core/views/error.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            "You do not have required permissions to access to this resource."
+        ),
+    },
+    404: {
+        "title": _("Page not found"),
+        "content": _("The requested resource does not exists."),
+    },
+    500: {
+        "title": _("Server bad request"),
+        "content": _("Something went wrong. Please retry later."),
```
Comment (sampaccoud) [inline src/richie/apps/core/views/error.py:22]: I think retry is used in IT only.
```suggestion
        "content": _("Something went wrong. Please try again later."),
```

File: src/richie/locale/es_ES/LC_MESSAGES/django.po [hunk 9]
```diff
@@ -2,7 +2,7 @@ msgid ""
 msgstr ""
```
Comment (sampaccoud) [inline src/richie/locale/es_ES/LC_MESSAGES/django.po:2]: Note that you will still need to enter these translations in Crowdin as populated .po files are pulled from there when building docker images.

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**